src/Control/Distributed/Process/Tests/Closure.hs

```diff
@@ -363,19 +378,26 @@ testSpawnSupervised TestTransport{..} rtable = do
     [node1, node2]       <- replicateM 2 $ newLocalNode testTransport rtable
     [superPid, childPid] <- replicateM 2 $ newEmptyMVar
     thirdProcessDone     <- newEmptyMVar
+    linkUp               <- newEmptyMVar
 
-    forkProcess node1 $ do
+    p <- forkProcess node1 $ do
       us <- getSelfPid
       liftIO $ putMVar superPid us
-      (child, _ref) <- spawnSupervised (localNodeId node2) (waitClosure 1000000)
-      liftIO $ do
-        putMVar childPid child
-        threadDelay 500000 -- Give the child a chance to link to us
-        throw supervisorDeath
+      (child, _ref) <- spawnSupervised (localNodeId node2)
+                                       (sendPidClosure us `seqCP` $(mkStaticClosure 'expectUnit))
+      _ <- expect :: Process ProcessId
+
+      liftIO $ putMVar childPid child
+      -- Give the child a chance to link to us
+      _ <- expect :: Process ()
+      throw supervisorDeath
 
     forkProcess node2 $ do
       [super, child] <- liftIO $ mapM readMVar [superPid, childPid]
       ref <- monitor child
+      -- because monitor message was sent before message to process
+      -- we hope that it will be processed before
+      send p ()
```

Comment:
I think Tim expected this to be doable with the management interface but the monitoring event was not yet implemented.